docs/docs/05-developer-guide/09-grpc/40-advanced/00-await-workflow-events.md

```diff
@@ -0,0 +1,74 @@
+# Await Workflow Events
+
+You can await a `WorkflowEvent` using the `AwaitWorkflowEvent` RPC call.
+
+## `AwaitWorkflowEventRequest`
+
+You need to create an `AwaitWorkflowEventRequest` to do so. The protobuf definition is:
+
+```proto
+message AwaitWorkflowEventRequest {
+  WfRunId wf_run_id = 1;
+  repeated WorkflowEventDefId event_def_ids = 2;
+  repeated WorkflowEventId workflow_events_to_ignore = 3;
+}
+```
+
+### Required Parameters
+
+The three required values are:
+
+1. `wfRunId` is the ID of the `WfRun` that the `WorkflowEvent` is thrown from.
+2. `eventDefIds` is a repeated field including the IDs of the `WorkflowEventDef`s you want to await thrown `WorkflowEvent`s for. The request will return the first matching `WorkflowEvent` thrown. If this field is empty, the request will return the first `WorkflowEvent` thrown by the `WfRun`.
+3. `workflowEventsToIgnore` is a repeated field of IDs of `WorkflowEvents` that you want to ignore. This gives the client the ability to ignore `WorkflowEvent`s that have already been awaited. See [Ignoring WorkflowEvents](#ignoring-workflowevents) for an example.
+
+## Examples
+
+### Awaiting a single `WorkflowEvent`
+
+```java
+WorkflowEvent event = client.awaitWorkflowEvent(AwaitWorkflowEventRequest.newBuilder()
+            .setWfRunId(WfRunId.newBuilder().setId("your-workflow-run-id"))
+            .addEventDefIds(WorkflowEventDefId.newBuilder().setName("my-workflow-event-def"))
+            .build());
+```
+
+### Using Deadlines
+
+Upon the execution of a `THROW_EVENT` node, LittleHorse will always ensure that your `WorkflowEvent`s are thrown and returned to any clients awaiting them. However, you may still find it useful to set a [gRPC deadline](https://grpc.io/docs/guides/deadlines/) on your `AwaitWorkflowEvent` request in case a `WorkflowEvent` is not thrown within a specified period of time.
+
+```java
```

Comment:
This PR is already a ton of work, but we will need to add `<Tabs>` and support the other two languages.

Reply:
Agreed, do you think we could get this PR out of the way and add an issue for adding other language support?